File: bmad_method/llm_connectors/ollama_client.py
import os
import requests
import json
from .base_client import BaseLlmClient

class OllamaClient(BaseLlmClient):
    def __init__(self):
        super().__init__()
        self.base_url = os.getenv('OLLAMA_BASE_URL')
        self.model = os.getenv('OLLAMA_MODEL')
        self.url = f"{self.base_url}/api/generate"

    def generate_plan(self, system_prompt: str, user_prompt: str) -> str:
        payload = {
            "model": self.model,
            "system": system_prompt,
            "prompt": user_prompt,
            "format": "json",
            "stream": False
        }
        response = requests.post(self.url, json=payload)
        response.raise_for_status() # Raise an exception for bad status codes
        return response.json()['response']

File: Godot-MCP/server/src/llm-connectors/ollama_client.ts
import fetch from 'node-fetch';
import { BaseLlmClient } from './base_client';

export class OllamaClient implements BaseLlmClient {
    private readonly baseUrl: string;
    private readonly model: string;
    private readonly url: string;

    constructor() {
        this.baseUrl = process.env.OLLAMA_BASE_URL || 'http://localhost:11434';
        this.model = process.env.OLLAMA_MODEL || 'llama3';
        this.url = `${this.baseUrl}/api/generate`;
    }

    async generatePlan(systemPrompt: string, userPrompt: string): Promise<string> {
        const payload = {
            model: this.model,
            system: systemPrompt,
            prompt: userPrompt,
            format: 'json',
            stream: false,
        };

        const response = await fetch(this.url, {
            method: 'POST',
            body: JSON.stringify(payload),
            headers: { 'Content-Type': 'application/json' },
        });

        if (!response.ok) {
            const errorBody = await response.text();
            throw new Error(`Ollama API request failed with status ${response.status}: ${errorBody}`);
        }

        const jsonResponse = await response.json() as { response: string };
        return jsonResponse.response;
    }
}
